design/one-pager-simple-class-selection.md

```diff
@@ -0,0 +1,411 @@
+logical grouping of infrastructure resources such as providers, resource
+classes, and managed resources. These namespaces might group infrastructure by
+its geographical region, or its environment (e.g. "production"), distinct from
+application namespaces. This potentially many to many application to
+infrastructure namespace pattern - currently established by portable resource
+classes - makes it difficult to mentally model the relationship between
+applications and the infrastructure that supports them.
+
+It is an uncommon pattern in Kubernetes to namespace infrastructure resources.
+Existing infrastructure kinds such as nodes, persistent volumes, and storage
+classes are cluster scoped concepts. This document proposes Crossplane Services
+resources (with the exception of resource claims) become cluster scoped. This
+provides a single, easy to find scope in which resource claim authors can
+discover the resource classes available to them, and the managed resources
+produced by their claim. The logical groupings provided by infrastructure
+namespaces that we frequently think of as “production”, “development”, or
+“east-coast” can be modelled using labels.
+
+### Exact Matching and Static Provisioning
+
+Resource claims would retain their existing `.spec.classRef` field under this
+proposal. In fact the `.spec.classSelector` would be used to set the `classRef`,
+just like Crossplane's [workload scheduling]. Resource claim authors who desired
+dynamic provisioning but had a specific resource class in mind could bypass the
+label selection process by explicitly referencing a resource class, for example:
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  namespace: acme-team
+  name: gitlab-database
+spec:
+  classRef:
+    apiVersion: database.gcp.crossplane.io/v1alpha2
+    kind: CloudSQLInstance
+    namespace: us-east-1
+    name: gitlab-experimental
+  writeConnectionSecretToRef:
+    name: app-postgresql-connection
+```
+
+Static provisioning is unaffected by this proposal; setting the
+`.spec.resourceRef` field of a resource claim would continue to bypass resource
+classes entirely. The resource claim would attempt to bind to the explicitly
+referenced managed resource.
+
+### Rudimentary Scheduling
+
+While not a goal of this proposal, it's worth noting that the label selector
+pattern enables rudimentary scheduling of resource claims to resource classes.
+Class and claim authors _could_ define a labelling schema that allowed claim
+authors to describe the desired properties of their claim using labels. The
+below example illustrates a labelling schema that could be used to describe a
+`RedisCluster` claim that needs a replicated Redis 3.2 instance running on the
+west coast of the USA:
+
+```yaml
+apiVersion: cache.crossplane.io/v1alpha1
+kind: RedisCluster
+metadata:
+  name: my-cache
+spec:
+  classSelector:
+    matchLabels:
+      engineVersion: "3.2"
```

Comment:
this is a very compelling aspect of labels/selectors and offers users a lot of flexibility in modeling different label schemas